framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -273,41 +278,40 @@ recorder_result_t MediaRecorderImpl::stop()
 	medvdbg("MediaRecorderImpl::stop()\n");
 	RecorderWorker& mrw = RecorderWorker::getWorker();
 	if (!mrw.isAlive()) {
-		return RECORDER_ERROR;
+		return RECORDER_ERROR_NOT_ALIVE;
 	}
-	mrw.enQueue(&MediaRecorderImpl::stopRecorder, shared_from_this(), 0);
 
+	mrw.enQueue(&MediaRecorderImpl::stopRecorder, shared_from_this(), RECORDER_OK);
 	return RECORDER_OK;
 }
 
-void MediaRecorderImpl::stopRecorder(int errcode)
+void MediaRecorderImpl::stopRecorder(recorder_result_t ret)
 {
 	medvdbg("stopRecorder mCurState : %d\n", (recorder_state_t)mCurState);
 
-	if (errcode == 0 && mCurState != RECORDER_STATE_RECORDING && mCurState != RECORDER_STATE_PAUSED) {
-		notifyObserver(OBSERVER_COMMAND_ERROR);
+	if (ret == RECORDER_OK && mCurState != RECORDER_STATE_RECORDING && mCurState != RECORDER_STATE_PAUSED) {
+		notifyObserver(OBSERVER_COMMAND_STOP_ERROR, RECORDER_ERROR_INVALID_STATE);
 		meddbg("stopRecorder Failed mCurState : %d\n", (recorder_state_t)mCurState);
 		return;
 	}
 
 	audio_manager_result_t result = stop_audio_stream_in();
 	if (result != AUDIO_MANAGER_SUCCESS) {
 		meddbg("stop_audio_stream_in failed ret : %d\n", result);
-		if (errcode == 0) {
-			/* TODO : inform device control fail */
-			notifyObserver(OBSERVER_COMMAND_ERROR);
-			return;
+		if (ret == RECORDER_OK) {
+			notifyObserver(OBSERVER_COMMAND_STOP_ERROR, RECORDER_ERROR_STOP_AUDIO_FAILED);
 		}
+		return;
```

Comment:
in "ret != RECORDER_OK" this case,  
The user does not get notify.
Is it the right logic?

Reply:
junsik is right, return must be moved to above bracket.